app/bundles/PluginBundle/Bundle/PluginBundleBase.php

```diff
@@ -33,28 +33,41 @@ public static function onPluginInstall(Plugin $plugin, MauticFactory $factory, $
     public static function installPluginSchema(array $metadata, MauticFactory $factory, $installedSchema = null): void
     {
         if (null !== $installedSchema) {
-            // Schema exists so bail
+            // Schema already exists, so no need to proceed
             return;
         }
 
         $db             = $factory->getDatabase();
         $schemaTool     = new SchemaTool($factory->getEntityManager());
         $installQueries = $schemaTool->getCreateSchemaSql($metadata);
 
-        $db->beginTransaction();
-        try {
-            foreach ($installQueries as $q) {
+        foreach ($installQueries as $q) {
+            // Check if the query is a DDL statement
+            if (self::isDDLStatement($q)) {
+                // Execute DDL statements outside of a transaction
                 $db->executeQuery($q);
+            } else {
+                // For non-DDL statements, use transactions
+                try {
+                    $db->beginTransaction();
+                    $db->executeQuery($q);
+                    $db->commit();
+                } catch (\Exception $e) {
```

Comment:
It sucks. I can see that the Doctrine methods suggests that they throw just an `Exception`. They should specify which one in more detail. From what I see, we should only catch the `DriverException` that the `doCommit()` method throws. Every other exception should not be caught.

cc @fedys please keep me honest here.

Reply:
I find it okay to catch the generic `\Exception` here as the exception is always re-thrown on [line 60](https://github.com/mautic/mautic/pull/13249/files#diff-576967f75fc32403e3fcd5d328593ee107e7795f3841c81c8ffa85aabc743cc0R60).

Reply:
I missed that. Ok then 👍